File: ts/s3.ts
import AWS from 'aws-sdk';

export class s3 {

  s3: Any;

  constructor() {
    AWS.config.update({region: 'ap-southeast-2'});

    this.s3 = new AWS.S3();
    this.bucket = 'BUCKET_NAME';
  }

  getObject(name: String): Promise<Any> {
    return new Promise<Any>((resolve, reject) => {
      const params = {
        Bucket: this.bucket,
        Key: name,
      };

      this.s3.getObject(params, (err, data) => {
        if (err) {
          console.log(err, err.stack);
          reject(err);
        } else {
          resolve(data.Body);
        }
      });
    });
  }

  putObject(body: Number, name: String): Promise<Any> {
    return new Promise((resolve, reject) => {
      const params = {
        Body: body,
        Bucket: this.bucket,
        Key: name,
      };

      this.s3.putObject(params, (err,data) => {
        if (err) {
          console.log(err, err.stack);
          reject(err);
        } else {
          resolve();
        }
      });
    });
  }
}

File: py/s3.py
import logging
logger = logging.getLogger(__name__)

import boto3
import os

class S3:
    '''
    A wrapper class for the Amazon s3 SDK
    '''
    def __init__(self):
        self.s3 = boto3.resource('s3')
        self.S3_NAME = 'BUCKET_NAME'

    def getObject(self, name):
        try:
            object = self.s3.Object(self.S3_NAME, name)
            return object.get()
        except Exception as error:
            logger.error(error)
            raise

    def putObject(self, body, name):
        try:
            object = self.s3.Object(self.S3_NAME, name)
            object.put(Body=body)
        except Exception as error:
            logger.error(error)
            raise
